docs/authoring-templates.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "Authoring Templates"
+weight: 60
+---
+
+The AsyncAPI generator has been built with extensibility in mind. The package uses a set of default templates to let you generate documentation and code. However, you can create and use your own templates. In this section, you learn how to create your own one.
+
+To work on a template, you need an AsyncAPI specification file that you can use for testing. For this purpose, you can use [this](../test/docs/dummy.yml) dummy file as it's purpose is to cover as many features of AsyncAPI as possible. Do not copy this file to your template but use it directly from this repo like this: `ag https://raw.githubusercontent.com/asyncapi/generator/master/test/docs/dummy.yml ./your-template`
+
+> In case you find some features missing or other possible improvements in the dummy file, suggest changes. The goal is to build a file that all templates can use and check their specification features coverage.
+
+## Common assumptions
```

Comment:
```suggestion

```

Like I said before, this header isn't right for a docs site. 😀

Let's make sure we think about this one in a different way, such as the following solution I can help propose. I think you need to turn this into one of our `Remember` React components, such as the one we have in our [Getting Started: Servers](https://www.asyncapi.com/docs/tutorials/getting-started/servers) tutorial. Then, all you have to do is determine in which current section it makes most sense to be in. It could also be that you need to turn this list into more than one `Remember` React component, adding individual items to where they most apply.

<img width="1000" src="https://user-images.githubusercontent.com/19964402/176539331-7aed044d-1628-4fc8-9c91-e33bfb3a20a0.png">

Reply:
Totally forgot about this! I guess renaming the Common assumptions to Remember would fit the architecture as well🤔 And both "Remember" and "Common assumptions" kind of indicate the same thing, what'd you think @derberg @Florence-Njeri 😁

Reply:
`Common assumptions` was an unfortunate name from the very beginning, but `Remember` and note is also not very good in this case. This section explains in short some basic information about templates, like intro to features. We could later think on transforming it into a kind of FAQ like: 
- Q: `how can I write a code that performs some modifications in the template after generation or before generation process` 
- A: Please use hooks -> link 


anyway not in scope of this PR, just make a note 😉 

so for now, best, just remove the header, and `Special file names` turn into a separate file, and put it at the end of the list in the navigation

Reply:
@derberg can the common assumptions be considered as features of a template?

Reply:
kind of 😄 because it is a mix of features and common information....really bad section 😄 

sometimes it talks about features, sometimes just states facts, like that template files should be in `template` dir 😄

Reply:
@derberg @Florence-Njeri @pratik2315 

Nope, sorry kidos.. no Q/A here. Adding an FAQ section is a very simplistic and unfortunately lazy way to handle TW problems. 

We really have to think about an appropriate way to add this information and if it is not with a remember react component, then we still need to plan where in the outline this information fits 
💜👍🏽👍🏽

Reply:
No FAQ's!!